dubbo-go/test/expected.data.yml

```diff
@@ -0,0 +1,130 @@
+#
+# Copyright 2021 SkyAPM org
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+segmentItems:
+  {{- range .segmentItems }}
+    {{- if eq .serviceName "dubbo-go-server" }}
```

Comment:
Notice, take a look at this context, https://github.com/apache/skywalking/pull/7849#discussion_r719884471.

`range` should not be used in the expected file.

@arugal You also should update the existing cases.